### ISSUE TITLE
Support for older versions?

### ISSUE THREAD

**Issue by lightvector** [FloopCZ/tensorflow_cc#117]
This repo was awesome, I just had one issue using it - I was running on Ubuntu 16.04 rather than 18.04, and had only g++-5, rather than g++-7.

However, I also didn't see any reason why it wouldn't work if I simply reverted the change here: 9d4a253b5616a848240c6bc6f3d462f6992d853d, since it seemed to be precisely this recent commit that was requiring all of the things that were failing for me, and for no particularly-good-for-me reason as far as I could tell (e.g. if I had merely discovered this repo a month earlier I would have cloned a version without that commit).

So I reverted that change... and then everything worked. Just wanted to throw this out there in case there's a nice way to make this kind of thing more configurable.

**Comment by lightvector** [comment]
Thanks, but I tried compiling again, and this did NOT work. It got through part of the build, but failed on whatever happens after compile_linux_protobuf.sh:

    compile_linux_protobuf.sh finished successfully!!!
    /bin/sh: 1: g++-7: not found
    ../../platform/posix/make.common:61: recipe for target 'depend' failed
    make[3]: *** [depend] Error 127
    CMakeFiles/tensorflow_shared.dir/build.make:106: recipe for target 'tensorflow-stamp/tensorflow_shared-configure' failed
    make[2]: *** [tensorflow-stamp/tensorflow_shared-configure] Error 2
    CMakeFiles/Makefile2:104: recipe for target 'CMakeFiles/tensorflow_shared.dir/all' failed
    make[1]: *** [CMakeFiles/tensorflow_shared.dir/all] Error 2
    Makefile:127: recipe for target 'all' failed
    make: *** [all] Error 2

Apparently g++-7 is hardcoded in places that ignore GCC_HOST_COMPILER_PATH?

The only solution I have right now is to manually revert the revision I mentioned before attempting to compile, which feels hacky and not obviously sustainable if things continue to develop...

**Comment by lightvector** [comment]
(Maybe the answer is just to upgrade to g++-7 and going forward in the future just keep in sync with the version of g++ hardcoded in this repo?)

**Comment by FloopCZ** [comment]
Yeah, sorry for that. It is hardcoded [here](https://github.com/FloopCZ/tensorflow_cc/blob/9d4a253b5616a848240c6bc6f3d462f6992d853d/tensorflow_cc/cmake/TensorflowShared.cmake#L15).

The reason is that nsync can not be compiled with `g++-8`, which is the default `g++` command on Arch Linux and Tensorflow's contrib/makefile project hardcodes `g++` command in its compilation script. As long as this is the case. please remove the line that patches the nsync compilation file. Or better - update your compiler, on Ubuntu 18.04 and Arch Linux, it is quite simple from the official repositories. :-)

**Comment by lightvector** [comment]
No worries. I'm mainly running into it because (at least as of a few months ago) the default g++ installed when I spin up a new AWS machine with their 'machine learning' image is g++-5. But yeah, I can update g++. Thanks for your help!
